### PR TITLE
Task 5 done

### PR DESCRIPTION
**Pull request by Kernstall**

### CODE REVIEW

File: UI/resources/photoPostModule.js [hunk 1]
```diff
@@ -95,15 +100,233 @@ const POST_API = (function(){
...
+
+        function addZero(i) { //for date-formating purpose
+            if (i < 10) {
+                i = "0" + i;
+            }
+            return i;
+        }
+
+        function validatePhotoPost(photoPost) {
+            return photoPost && Object.keys(photoPost).length < 9 && typeof photoPost.id === "string" &&
```
Comment (tyndria) [inline UI/resources/photoPostModule.js]: too big return statement
the check `` Object.keys(photoPost).length < 9 `` is redundant
the minimal requirement for this function is to divide this return statement into several expressions. I mean, create variables isAuthorValid, isDescriptionValid, ..., and return expression ``isAuthorValid && isDescriptionValid && ... `` from this function

File: UI/resources/photoPostModule.js [hunk 2]
```diff
@@ -95,15 +100,233 @@ const POST_API = (function(){
...
+                feedList.appendChild(assemblePhotoPost(photoPost));
+           }
+
+        function drawFrontPostOnScreen(photoPost) {
+            feedList.insertBefore( assemblePhotoPost(photoPost) , feedList.firstChild);
+        }
+
+        function assemblePhotoPost(photoPost){
+            if( validatePhotoPost(photoPost)) {
+                var feedPost = document.createElement('section');
```
Comment (tyndria) [inline UI/resources/photoPostModule.js]: You are supposed to use template for creating photoPost. Your option is applicable, but it's better to use template to do less work. The info about templates should be in the lectures

File: UI/resources/photoPostModule.js [hunk 3]
```diff
@@ -194,11 +460,61 @@ const POST_API = (function(){
...
+                }
+                var likeMask = document.getElementById(id).getElementsByClassName('likeMask liked')[0];
+                var likeCount = document.getElementById(id).getElementsByClassName('likeCount')[0];
+                if(likeMask!=undefined) {
+                    likeMask.setAttribute('class', 'likeMask');
+                    if(likeCount!=undefined) {
+                        likeCount.textContent = Number(likeCount.textContent)-1;
+                    }
+                }else{
+                    let likeMask = document.getElementById(id).getElementsByClassName('likeMask')[0];
```
Comment (tyndria) [inline UI/resources/photoPostModule.js]: It's better:
not to mix var and let (you should know difference between var, let and const)
not to mix arrow function and function expression (you should also understand the difference, read about that)

File: UI/resources/photoPostModule.js [hunk 4]
```diff
@@ -194,11 +460,61 @@ const POST_API = (function(){
...
+        },
+        
+        likePost: function (id) {
+            if(username == undefined){
+                return;
+            }
+            var likedPost = photoPosts.find(function(post){
+                return post.id===id;
+            });
+            if(likedPost!= undefined ){
```
Comment (tyndria) [inline UI/resources/photoPostModule.js]: !==

File: UI/resources/photoPostModule.js [hunk 5]
```diff
@@ -158,14 +401,15 @@ const POST_API = (function(){
         editPhotoPost: function (id, photoPost) {
```
Comment (tyndria) [inline UI/resources/photoPostModule.js]: I have error in console, when call 
``POST_API.editPhotoPost(100, {description: 'NEW DESCRIPTION'});``

**Comment by tyndria** [comment]
If user haven't logged in yet, you should hide edit/delete buttons on the post

**Comment by Kernstall** [comment]
@tyndria Fixed problemes you mentioned

**Pull request merged**